refactor: header redesign

wesleyboar opened the pull request:
## To Do

> **Warning**
> The `main` branch has changed drastically since work on header redesign began.
> - [ ] resolve merge conflicts **and** port all work to [Core-Styles which now has a demo](https://github.com/TACC/Core-Styles/pull/258)

<details><summary>✓ Before Small-Screen UI Design</summary>

- [x] large-screen
	- [x] support on unedited Portal and Docs
	- [x] fix bugs
- [x] small-screen (designed by dev)
	- [x] design (by developer)
	- [x] implement design
	- [x] support on unedited Portal and Docs
	- [x] fix bugs

</details>
<details><summary>✓ After Small-Screen UI Design</summary>

- [x] small-screen
	- [x] re-style menu toggles
	- [x] position menu toggles
	- [x] position menu content
	- [x] re-style menu content

</details>
<details open><summary>→ After Small-Screen UI Design Review</summary>

- [ ] small-screen
	- [ ] implement UI/UX changes
	- [ ] fix bugs

</details>
<details open><summary>→ Clean Up Code</summary>

- [ ] abandon support for unedited Portal and Docs
	- [x] small-screen menu toggles
	- [ ] small-screen anything else
	- [ ] large-screen anything else
- [ ] remove `gh-101` class and usage
- [ ] polish so component-ization can be pushed off until later

</details>
<details open><summary>→ Merge in Sub-Branches</summary>

- `task/GH-101-header-redesign`
	- [ ] `task/GH-101-header-redesign--mobile-nav-layout-2`

</details>
<details open><summary>… Clean Up PR Desc</summary>

- [ ] update screenshots
- [ ] review "Notes"

</details>





## Overview

Redesign the header.





## Related

- ~~GH-101~~ → [~~FP-1458~~](https://jira.tacc.utexas.edu/browse/FP-1458) → [**CMD-117**](https://tacc-main.atlassian.net/browse/CMD-117)
- requires #192
- requires https://github.com/TACC/Core-Portal/pull/395
- requires [__taccaci/frontera-tech-docs !20__](https://bitbucket.org/taccaci/frontera-tech-docs/pull-requests/20/task-gh-101-header-redesign)
- expects https://github.com/TACC/Core-CMS/pull/824

<details>
<summary>Note to Self</summary>

- Estimated Completion Time: 2 weeks
- Actual Completion Time: 4+ weeks

This difference is almost made up for by #264. Progress has been delayed by other work.

</details>





## Changes

- #192
- Redesign.
- Support more future-compatible CSS features.
- Add more global `env()` variables.
    - Remove "freeze variables" hack.
- Add new CSS components, utilities, and mixins.
- Document.

---

<details><summary>all pull requests</summary>

- #192
    - #221
    - #224
- #220
- #225
- #226
- #227
- #228
- #229
- #232
- #237
- #238
- #239
- #241
- \* #242
- #244
- \* #247
- \* #248

> \* fixes & polish

</details>
<details><summary>complete changelog</summary>

- __New__: Add _more_ future-compatible CMS support:
    - [`:not()` with multiple selectors](https://preset-env.cssdb.org/features#not-pseudo-class) (fff0a3d)
    - [nesting selectors](https://preset-env.cssdb.org/features#nesting-rules) (dc4f7db)
    - [custom selectors](https://preset-env.cssdb.org/features#custom-selectors) (c988a7b)
- __New__: Import `env()` vars for _all_ themes (ea43ec9)
- __New__: Install `merge-lite` (ea43ec9)
- __New__: Remove "freeze variables" (faab92b, 54635b6)
    - new searchbar design does not need them
    - new `env()` support replaces them
- _Minor_: Add misc. helpful comments
- _Minor_: Add more CSS plugin tests (ebeebe5, dc4f7db)
- __New__: Add new components:
    - `c-branding`
    - `c-logo`
- __New__: Redesign Search bar
- __New__: Backwards compatibility for `backdrop-filter` (dc4f7db)
- __New__: Add `@custom-media` for navigation breakpoints (59a97d3)
- __New__: Add new mixins (in "tools"):
    - `x-arrow`
    - `x-hide`
    - `x-mobile-button`
- __Doc__: Add missing Readme for trumps
- __New__: Add docs-specific and portal-specific header-dependent stylesheets
- __New__: Redesign Portal and CMS navs
- _Minor_: Update `s-portal-nav` sample markup (to reflect future Portal markup)
- __New__: Add new utilities (in "trumps"):
    - `u-hide`
- __Doc__: Document themes
- _Minor_: Import all icon styles
- __Doc__: Describe site assets required by subsites
- __New__: Add `[data-status="loaded"]` after Portal nav load (for styling) (e38594d)
- _Minor_: Remove Frontera-specific stylesheets
- _Untracked_: Changes __after__ 1248117.

</details>




## Screenshots

> ⚠️ Out of date!

_Hover over an image to see a description (on desktop browser)._

<details><summary><strong>Wide Screen (1200px)</strong></summary>

> ⚠️ Out of date!

<dl>

<dt>Docs (Logged <u>Out</u>)</dt>
<dd>

(No State)
<img
title="Docs [Wide] (No Auth)"
  alt="Docs [Wide] (No Auth)" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714434-6cdf8500-caa3-11eb-8603-b816dce229b7.png" />
</dd>

<dt>CMS (Logged <u>Out</u>)</dt>
<dd>

CMS Nav: Link (Hover) | Dropdown (Open) | Dropdown Item/Link (Hover)
<img
title="CMS [Wide] (No Auth) - CMS Nav - Link Hover"
  alt="CMS [Wide] (No Auth) - CMS Nav - Link Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714264-4a4d6c00-caa3-11eb-9cd0-715e0cbf0138.png" /><img
title="CMS [Wide] (No Auth) - CMS Dropdown - Open"
  alt="CMS [Wide] (No Auth) - CMS Dropdown - Open" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714258-491c3f00-caa3-11eb-9b2e-6453c2ee7ced.png" /><img
title="CMS [Wide] (No Auth) - CMS Dropdown - Item Hover"
  alt="CMS [Wide] (No Auth) - CMS Dropdown - Item Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714252-47527b80-caa3-11eb-9641-6d6b5877e276.png" />
</dd>

<dt>Portal (Logged <u>In</u>)</dt>
<dd>

Portal Nav: Closed | Link (Hover) | Open
<img
title="Portal [Wide] (Authed)"
  alt="Portal [Wide] (Authed)" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714437-6ea94880-caa3-11eb-85cb-4db43f1839a3.png" /><img
title="Portal [Wide] (Authed) - Portal Nav - Link Hover"
  alt="Portal [Wide] (Authed) - Portal Nav - Link Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714441-70730c00-caa3-11eb-9644-d97937f550c7.png" /><img
title="Portal [Wide] (Authed) - Portal Nav - Open"
  alt="Portal [Wide] (Authed) - Portal Nav - Open" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714445-71a43900-caa3-11eb-841d-aa3bde6ac423.png" />
</dd>

<dt>Portal (Logged <u>Out</u>)</dt>
<dd>

Portal Nav: Closed | Link (Hover)
<img
title="Portal [Wide] (No Auth)"
  alt="Portal [Wide] (No Auth)" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714449-72d56600-caa3-11eb-91b7-fd1ff957d068.png" /><img
title="Portal [Wide] (No Auth) - Portal Nav - Link Hover"
  alt="Portal [Wide] (No Auth) - Portal Nav - Link Hover" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714458-749f2980-caa3-11eb-9129-649e112e25dd.png" />
</dd>

</dl>
</details>



<details><summary><strong>Narrow Screen (768px)</strong></summary>

> ⚠️ Out of date!

<dl>

<dt>Docs (Logged <u>Out</u>) — CMS Nav & Login Link & Search Field</dt>
<dd>

(No State)
<img
title="Docs [Narrow]"
  alt="Docs [Narrow]" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714354-5cc7a580-caa3-11eb-980c-d3b188d6b09e.png" />

Nav Open: Close Button (Hover) | CMS Link (Hover)
<img
title="Docs [Narrow] - Nav - Open"
  alt="Docs [Narrow] - Nav - Open" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714364-5e916900-caa3-11eb-9b92-b57987b0633f.png" /><img
title="Docs [Narrow] - Nav - Open - Close Hover"
  alt="Docs [Narrow] - Nav - Open - Close Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714370-5fc29600-caa3-11eb-8d8e-a2e4b1b63573.png" /><img
title="Docs [Narrow] - Nav - Open - CMS Nav - Link Hover"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Link Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714412-681ad100-caa3-11eb-9cea-72bfa5b991cc.png" />

Dropdown Open: CMS Link (Hover) | CMS Dropdown Item/Link (Hover)
<img
title="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Link Hover"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Link Hover" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714391-64874a00-caa3-11eb-9bdc-070b3d3400de.png" /><img
title="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Item Hover"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Item Hover" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714384-63561d00-caa3-11eb-985e-166ffc0b5329.png" />

Dropdown Open: Login Link (Hover) | Search Field (Focus) | Search Field (Input)
<img
title="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Login Hover"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Login Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714420-694bfe00-caa3-11eb-8f7a-46645a24ac41.png" /><img
title="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Search Focus"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Search Focus" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714425-6b15c180-caa3-11eb-9676-3c66220f56e5.png" /><img
title="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Search Input"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open & Search Input" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714403-66510d80-caa3-11eb-8380-eca48cb376b8.png" />

Dropdown Open
<img
title="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open"
  alt="Docs [Narrow] - Nav - Open - CMS Nav - Dropdown Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714377-618c5980-caa3-11eb-80ff-6e87c52f576c.png" />
</dd>

<dt>CMS (Logged <u>In</u>) — Portal Nav</dt>
<dd>

Nav Open
<img
title="CMS [Narrow] - Nav - Open"
  alt="CMS [Narrow] - Nav - Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714301-520d1080-caa3-11eb-9a06-db41acf9d080.png" />

Nav Open: Link (Hover) | Dropdown (Open) & Link (Hover) | Dropdown (Open) & Dropdown Item/Link (Hover)
<img
title="CMS [Narrow] - Nav - Open - Portal Nav - Link Hover"
  alt="CMS [Narrow] - Nav - Open - Portal Nav - Link Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714333-576a5b00-caa3-11eb-8fd4-8704ea001650.png" /><img
title="CMS [Narrow] - Nav - Open - Portal Nav - Dropdown Open & Link Hover"
  alt="CMS [Narrow] - Nav - Open - Portal Nav - Dropdown Open & Link Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714330-56392e00-caa3-11eb-9170-9bc59fe26e30.png" /><img
title="CMS [Narrow] - Nav - Open - Portal Nav - Dropdown Open & Item Hover"
  alt="CMS [Narrow] - Nav - Open - Portal Nav - Dropdown Open & Item Hover" width="33%"
    src="https://user-images.githubusercontent.com/62723358/121714319-546f6a80-caa3-11eb-98e6-30f293bce1f5.png" />

Nav Open: Dropdown (Open)
<img
title="CMS [Narrow] - Nav - Open - Portal Nav - Dropdown Open"
  alt="CMS [Narrow] - Nav - Open - Portal Nav - Dropdown Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714311-533e3d80-caa3-11eb-86cc-1b1f38d4d560.png" />
</dd>

</dl>
</details>



<details><summary><strong>Minimum Width (320px)</strong></summary>

> ⚠️ Out of date!

<strong>CMS</strong>

CMS does respect cross-site minimum width.

<dl>
<dt>At Minimum Width</dt>
<dd>

Nav Open
<img
title="CMS [Min Width] - Nav Open"
  alt="CMS [Min Width] - Nav Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714297-50dbe380-caa3-11eb-85e0-ef75f03a68e6.png" />
</dd>

<dt>Below Minimum Width</dt>
<dd>

Nav Open
<img
title="CMS [Below Min Width] - Nav Open"
  alt="CMS [Below Min Width] - Nav Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714276-4b7e9900-caa3-11eb-888f-058e61200a65.png" /><img
title="CMS [Below Min Width] - Nav Open - Scroll"
  alt="CMS [Below Min Width] - Nav Open - Scroll" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714294-4faab680-caa3-11eb-851d-7163e1bb7a3e.png" />

Nav Open: Portal Nav
<img
title="CMS [Below Min Width] - Nav Open - Portal Nav - Open"
  alt="CMS [Below Min Width] - Nav Open - Portal Nav - Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714282-4d485c80-caa3-11eb-84c8-fcd4ba1a43d4.png" /><img
title="CMS [Below Min Width] - Nav Open - Portal Nav - Open - Scroll"
  alt="CMS [Below Min Width] - Nav Open - Portal Nav - Open - Scroll" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714290-4e798980-caa3-11eb-88cc-c188d370c9bb.png" />
</dd>
</dl>

<strong>Docs</strong>

Docs does *__not__ yet* respect cross-site minimum width.

_Docs can respect cross-site min width. I have temporary solution for now._

<dl>

<dt>At Minimum Width</dt>
<dd>

Nav Open
<img
title="Docs [Min Width] - Nav Open"
  alt="Docs [Min Width] - Nav Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714346-5b967880-caa3-11eb-9867-ae79af51f6ba.png" />
</dd>

<dt>Below Minimum Width</dt>
<dd>

Nav Open
<img
title="Docs [Below Min Width]"
  alt="Docs [Below Min Width]" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714336-589b8800-caa3-11eb-981a-6cd46c5e4bf0.png" /><img
title="Docs [Below Min Width] - Nav Open"
  alt="Docs [Below Min Width] - Nav Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121714338-59ccb500-caa3-11eb-986d-f5fd5f303505.png" />
</dl>

<strong>Portal</strong>

Portal does *__not__* respect cross-site minimum width.

_Portal can __not easily__ respect cross-site min width until a mobile design._

<dl>
<dt>At Minimum Width</dt>
<dd>

Nav Open
<img
title="Portal [Min Width] - Nav Open"
  alt="Portal [Min Width] - Nav Open" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121716691-f1330780-caa5-11eb-8538-0858f423f958.png" /><img
title="Portal [Min Width] - Nav Open - Scroll"
  alt="Portal [Min Width] - Nav Open - Scroll" width="50%"
    src="https://user-images.githubusercontent.com/62723358/121716721-fbed9c80-caa5-11eb-9c37-1131cff00223.png" />
</dd>
</dl>

</details>





## Testing

1. Set up testing environment using method from [Test CMS Changes](https://github.com/TACC/Core-CMS/wiki/Test-CMS-Changes). Specifically,

    1. `npm run settings` (new step, from #192)
    2. `npm ci`|`install`
    3. `npm run build`
    4. collect static files

2. Test these elements are present, functional, and match design on the matching subsites:

    <details><summary>Table</summary>

    | Element&nbsp;→<br />Subsite&nbsp;↓ | CMS Nav (Dropdown) | CMS Nav (Main Links) | Branding Bar | Logo | Search | Portal Nav (Logged In) | Portal Nav (Logged Out) |
    |-|-|-|-|-|-|-|-|
    | CMS&nbsp;(sans&nbsp;Portal) | ✓ | ✓ | ✓ | ✓ | | | |
    | CMS&nbsp;(with&nbsp;Portal) | ✓ | ✓ | ✓ | ✓ | | | |
    | Portal                     | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ |
    | Docs                      | ✓ | ✓ | ✓ | ✓ | ~ | ~ | ~ |

    Legend:
    - ✓ Has
    - ~ Has (but only because Frontera has portal and is only project with new docs)

    </details>

3. Test these elements have these layouts occur at the matching breakpoints on the matching subsites:

    <details><summary>Table</summary>

    | Subsite | Breakpoint&nbsp;→<br />Elements&nbsp;↓ | 320px– | 1200px– | 1200px+ | 1200px—1400px | 1400px—1680px | 1680px+ |
    |-|-|-|-|-|-|-|-|
    | CMS&nbsp;(sans&nbsp;Portal) | CMS Nav            | ┣━ | __vert.__ list | __horz.__ list at __right__ | __some__ space between links | __more__ space between links | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | CMS Nav            | ┣━ | __vert.__ list | __horz.__ list at __left__ | ┣━ | __some__ space between links | __more__ space between links |
    | CMS,&nbsp;Portal,&nbsp;Docs | CMS Nav Bkgd       | ┣━ | translucent | opaque | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Portal Dropdown    | ┣━ | __wide__ list | __narrow__ list | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Portal Nav Toggle  | ┣━ | __narrow__ layout | __wide__ layout | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Nav Bar            | ┣━ | __large__ text | __small__ text | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Menu Icon          | __flex.__ width | __static__ width | __absent__ | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Branding Bar       | ┣━ | __short__ height | __taller__ height | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Nav Bar            | ┣━ | __short__ height | __taller__ height | ━━ | ━━ | ━┫ |
    | CMS,&nbsp;Portal,&nbsp;Docs | Search Bar         | ┣━ | __taller__ height | __short__ height | ━━ | ━━ | ━┫ |

    </details>

4. Ensure dropdown icon is:

    <details><summary>List</summary>

    - `⌄` when dropdown is __open__
    - `^` when dropdown is __closed__
    - __bigger__ for CMS Nav at __1200px–__
    - __smaller__ for CMS Nav at __1200px+__
    - __consistent__ size for Portal Nav __above and below 1200px__

    </details>

## Notes

### Known Issues

1. <details>
    <summary>On mobile nav, the dropdown toggle ("chevron") is not thick enough.</summary>

    I am forgoing adding complexity in code to solve this, because there should be a Cortal icon for this shape.

    </details>
2. <details>
	<summary>Stand-alone CMS navbar collapses for a wider screen than before.</summary>

	- Before, on a screen of < 992px, navbar would collapse.
	- Now, on a screen of < 1200px, navbar will collapse.

	A CMS that belongs to a Portal is not affected, because it already behaves like this.

    </details>

## Known Deviations From Design

### Expected to Remain

1. <details><summary>Some color(s) of Search Bar use a standard color instead of color in design.</summary>

    I have prior private permission to make these "fixes" during translation. I plan (eventually) to teach Portal devs how to do the same.

    </details>

### Mobile Portal Nav Menu

If the mobile portal nav menu were moved to outside the navbar dropdown, then these likely would all be resolved. But Design has agreed to let me do that later.

2. <details>
    <summary>On narrow screen, <em>Search Bar</em> height and <em>Search Bar + Portal Nav link</em> height are shorter than design.</summary>

    I just didn't want to maintain another value when the two were so close—and sometimes Design is close when they mean to match. If this is noticed as a bug, then I will fix it.

    </details>
3. <details>
	<summary>On medium and wide screen, space left of Logo and right of Portal Nav do not match design.</summary>

	- Before, Design said space left of Logo should match space left of Portal sidebar icon.
	- Now, Design says left of logo space should be 30px __and__ match space right of portal nav.
	- Also, the width appears static and too wide (see Known Deviation 5).

	So, I ignored 30px and made spaces align because my experience shows me that aligning and consistency is Design's goal here, not an exact pixel value.

    </details>
4. <details>
	<summary>On narrow screen, Portal Nav dropdown is pushing down CMS Nav.</summary>

    This is a stopgap to __not__ have the Portal Nav dropdown fall over CMS Nav, but __not__ yet move it to the Designer-suggested location (which is a deviation form design, see Known Deviation 6).

    </details>
5. <details>
	<summary>On narrow screen, Portal Nav link is wider than necessary (for short usernames)</summary>

	- To (A) place the Portal Nav link and Search Bar on one row _and_ (B) have Portal Nav dropdown push down CMS Nav _but_ (C) _not_ reinvent Bootstrap functionality, the Portal Nav link must have a fixed width.
    - The fixed width for the Portal Nav link is the maximum needed to fit the widest username (8-characters, all `m`'s).

    </details>
6. <details>
	<summary>On narrow screen, Portal Nav location is not deviating from design as Designer suggested it should.</summary>

    - Design has said the Portal Nav link should (deviate from design) be on the Navbar to the right of the close button.
    - The current location is achieved using code that may support the move to the desired location.
    - The change is major and I am already well beyond time estimated for this task.

    </details>

wesleyboar committed:
GH-101: Branding Bar

- Migrate `branding_logos` CSS to `c-branding` & `c-logo`.
- In `s-header` style `.c-logo` not `.navbar-brand`.
- Update and format markup for `header_branding` and `header_logo`.
- Comment on now-unused secrets values.

wesleyboar committed:
Merge pull request #221 from TACC/task/GH-101-header-redesign--env-variables

wesleyboar committed:
Merge branch 'task/GH-101-header-redesign' into task/GH-101-header-redesign--branding-bar

wesleyboar committed:
Merge pull request #220 from TACC/task/GH-101-header-redesign--branding-bar

GH-101: Branding Bar

wesleyboar committed:
GH-101: Navigation Bar (Partially Complete)

- Remove now-moot taccsite_custom:frontera-cms search bar font sizes.
- Support CSS custom selectors (known external bug).
- Test CSS custom selectors (known external bug).
- Remove outdated comment from `c-branding`.
- Set `c-logo` image height in component stylesheet.
- Add TODO for `html { font-size: 62.5%; }`
- Add arrow "mixin" (note: using `%-arrow` syntax).
- Redesign header (but not dropdowns and not portal nav).
- Remove now-moot `s-portal-nav` styles.
- Add & Update header env variables for themes.
- Perform "GH-129: Do Not Support CMS Menu Nav Link & Dropdown".

wesleyboar committed:
GH-101: Navigation Bar (Fix CMS Nav Bug in Docs)

Prevent dropdown link from being flagged as active.

wesleyboar committed:
GH-101: Navigation Bar (Fix Color Bug/s in Docs)

- (Real Fix) Ensure colors are available to Docs.
- (Possible Fix) Ensure correct colors are available to Portal.\*

\* The Portal has its own settings/color.css that is almost identical.

wesleyboar committed:
GH-101: Navigation Bar (Fix More Bugs in Docs)

- Load new styles solely for docs header.
- Force branding styles on old markup.
- Migrate subsite selectors from `trumps/s-header` to `tools/selectors`.
- Rename `:--in-__subsite__` selectors to `:--from-__subsite__`.
- Create new styles solely for docs header.
- Document "For (Other Sites)" i.e. `trumps/for-docs.*.css` stylesheets.

wesleyboar committed:
GH-101: Navigation Bar (Fix Img Max-Width in Docs)

Unset Docs' `img { max-width; }`. Docs does this too but for old markup.

wesleyboar committed:
GH-101: Navigation Bar (Fix & Doc Height Bugs)

wesleyboar committed:
GH-101: Navigation Bar (Set Dropdown Font Size)

wesleyboar committed:
GH-101: Navigation Bar (Use Constants for Header)

- Install `merge-lite` for deep merge constants and theme data.
- Use env() vars for header height.
- Remove unnecessary code and improve comments.
- Create constants for theme JSON.
- Merge constants on top of theme data.
- Update theme data comment syntax.

wesleyboar committed:
GH-101: Navigation Bar (Logo Width, CMS Nav Align)

- Add frontera-cms logo width overwrite (temp. ineffectual, see GH-223).
- Prevent Core (but not a project) from stretching logo too wide.
- Align CMS nav for narrow logos. Allow project-specific adaptation.

wesleyboar committed:
GH-101: Navigation Bar (Remove Redundant Style)

- Remove redundant color assignment for nav-link.

wesleyboar committed:
GH-101: Navigation Bar (Dropdowns & Constants)

- Add font weight constants.
- Add border constants.
- Style dropdowns and use new constants.

wesleyboar committed:
GH-191: Add prewatch (same task as prebuild).

wesleyboar committed:
Merge pull request #224 from TACC/task/GH-101-header-redesign--env-variables

Task/gh 101 header redesign  env variables (update)

wesleyboar committed:
GH-101: Navigation Bar (Fix Branding Bar Bug)

Use correct global var not wrong env var for branding bar bkgd.

wesleyboar committed:
GH-101: Navigation Bar (Avoid Dropdown Bug)

Scope dropdown styles to header so as to not effect any other dropdowns.

wesleyboar committed:
GH-101: Navigation Bar (Fix `.s-header` Usage)

1. Scope header-specific `.c-branding` styles with `.s-header`.
2. Clarify which `s-header` is being styled and why.
3. Remove `s-header` ID from sample header markup.
4. Add `s-header` class to `header-branding`.
5. Replace `s-header` ID with `header-navbar` ID.

wesleyboar committed:
GH-101: Navigation Bar (Fix Portal Menu Item Hue)

Portal needs to overwrite Bootstrap's color for active `dropdown-item`.

wesleyboar committed:
GH-101: Navigation Bar (Use Fallbacks for Vars)

Instead of defining a value to be overwritten, set a fallback.

This reduces one level of overwrite when tracking/debugging values.

wesleyboar committed:
GH-101: Navigation Bar (Consistent Line-Height)

Ensure Portal & Docs line-height (in header) matches CMS.

wesleyboar committed:
GH-101: Navigation Bar (Responsive Nav Item Width)

- The responsive space between nav items now considers portal elements.
- Whether header is loaded on portal is noted by presence of a class.

wesleyboar committed:
Merge branch 'task/GH-101-header-redesign' into task/GH-101-header-redesign--navigation-bar

wesleyboar committed:
Merge pull request #225 from TACC/task/GH-101-header-redesign--navigation-bar

wesleyboar committed:
GH-101: taccsite_custom: Frontera: Navigation Bar

https://github.com/TACC/Core-CMS-Resources/pull/47

wesleyboar committed:
Quick: GH-101: Fix font env vars

Without quotes, values were not recognized, and thus not applied.

wesleyboar committed:
GH-101: taccsite_custom: Mostly Revert its 64a5d03

wesleyboar committed:
GH-101: Portal Nav

- Use new portal env() vars for nav padding.
- No custom logo width (related to portal nav via navbar padding).
- Remove border between search and portal nav.
- Add Portal nav styles.
- Add new portal env() vars for nav padding.
- Comment on header link bkgd color env() var value.
- Add new header env() var for portal nav bkgd color.

wesleyboar committed:
Merge pull request #226 from TACC/task/GH-101-header-redesign--portal-nav

wesleyboar committed:
Quick: GH-101: Fix border color. Drop unused var.

wesleyboar committed:
Quick: GH-101: Add "nav state" media queries vars

wesleyboar committed:
GH-101: Header (Use .c-logo for Brand Bar Images)

- Remove branding images style that is made redundant.
- Allow logo height less than 100% (for navbar).
- Ensure navbar logo styles do not affect brand bar.
- Give limited flexibility to navbar logo.
- Use c-logo for branding images.

wesleyboar committed:
GH-101: Navigation Bar (Responsive Design)

- Add compressed and expanded styles in two files.
- Increase body min-width (to prevent wrap of navbar toggle).
- Remove condition for navbar breakpoint; now it's always 992px.
- Migrate some `s-header` CSS to `[…].expanded` and `[…].compressed`.
- Use `top` to position arrows isntead of `margin`.
- Truncate nav links.
- Style dropdowns.
- Add nav link text span wrappers (to prevent truncation of menu arrows).

wesleyboar committed:
Merge pull request #227 from TACC/task/GH-101-header-redesign--mobile-nav

wesleyboar committed:
GH-101: Navigation Bar (Mobile Menus & Toggle)

- Import all icon styles (not just some).
- Support mobile brandbar.
- Support mobile vs. desktop navbar & brandbar heights.
- Support mobile logo.
- Support mobile navbar collabsible menu.
- Add new theme constants for navbar & branbar heights.
- Add navbar menu button text.

wesleyboar committed:
Merge pull request #228 from TACC/task/GH-101-header-redesign--mobile-nav

GH-101: Navigation Bar (Mobile Menus & Toggle)

wesleyboar committed:
Quick: GH-101: Fix mobile navbar toggle visibility

wesleyboar committed:
Hack: GH-101: Mobile navbar to act like dropdown

wesleyboar committed:
Quick: GH-101: Comments on mobile nav padding vars

wesleyboar committed:
Hack: GH-101: Use Cortal icons sans markup change

- Tweak a comment (related cuz it's also icon hack).
- Migrate Portal nav styles to `s-portal-nav.css`.
- Add hack to swap FA icons with Cortal icons.
- Update CSS sample `s-portal-nav.html`.

wesleyboar committed:
Merge pull request #229 from TACC/task/GH-101-header-redesign--portal-icons-hack

Hack: GH-101: Use Cortal icons sans markup change

wesleyboar committed:
GH-101: Mobile (Search & Portal): Drop `ml-auto`

- Use CSS instead of markup to control margin left on header elements.
- Make header code DRY now that classNames match.

wesleyboar committed:
GH-101: Quick: Use custom selector for consistency

wesleyboar committed:
GH-101: Quick: Fix navbar toggle color

wesleyboar committed:
GH-101: Mobile (Search & Portal): New Layout

- Support `:not` with multiple elements.
- Layout Search and Portal elements in mobile nav.
- Edit existing relevant CSS to be consitent and accurately organized.

wesleyboar committed:
GH-101: Quick: Fix dropdown toggle size

The "chevron" was a bit too big. Fixed.

It is not thick enough on mobile, but the code should stay simple.

This code is all pending a real icon, anyway.

wesleyboar committed:
GH-101: Quick: Swap "Log Out" & "My Account" icons

wesleyboar committed:
GH-101: Quick: Icon Text for Portal/Docs Markup

wesleyboar committed:
Merge pull request #232 from TACC/task/GH-101-header-redesign--mobile-nav-layout

wesleyboar committed:
GH-101: Quick: Fix CMS Sel., Align Portal Nav Menu

wesleyboar committed:
GH-101: Quick: Ensure Footer is Behind Mobile Menu

wesleyboar committed:
Quick: Latest taccsite_custom code (frontera home)

wesleyboar committed:
GH-101: Hotfix: Padding Right for Portal Nav Link

wesleyboar committed:
GH-101: Hotfix: Prevent Portal Nav Link Underline

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
GH-101: Hotfix: Accurate Portal Nav Icons & Sizes

wesleyboar committed:
GH-101: Quick: Isolate Navbar Shrink & Truncate

Isolate shrinking and truncation of certain elements of the navbar.

wesleyboar committed:
GH-101: Hotfix: Title case all nav link text

wesleyboar committed:
Revert "GH-101: Hotfix: Title case all nav link text"

This reverts commit ba31e2e900e60e6b910fcc279f73a398efea4f56.

wesleyboar committed:
GH-101: Hotfix: Title case on "Log in" link text

wesleyboar committed:
GH-101: Quick: Fix unrealized specificity bug

wesleyboar committed:
GH-101: Portal Nav Transition / Animation Load

Load the portal nav into place, then transition it into normal width.

Give sidebar gentle push—not abruptly shove—when portal nav loads.

wesleyboar committed:
GH-101: Portal Nav Transition / Animation Comments

wesleyboar committed:
GH-101: Portal Nav Transition / Animation Migrate

Migrate the new code of the previous commits to a stylesheet, so Docs can load it.

wesleyboar committed:
Merge pull request #237 from TACC/task/GH-101-header-redesign--portal-nav-load-transition

wesleyboar committed:
GH-101: Portal Nav Transition / Animation Fixes

Ensure CMS & Docs and Portal load the correct styles for their need.

wesleyboar committed:
Merge pull request #238 from TACC/task/GH-101-header-redesign--portal-nav-load-transition

GH-101: Portal Nav (Load with Transition / Animation) - Bug Fixes / Missing Commit

wesleyboar committed:
Hotfix: GH-101: Portal Nav Visible Sans #101 Code

wesleyboar committed:
GH-101: Portal Nav Mobile - Link Style

- Limit large nav link padding to CMS links.
- Limit nav icon spacing depending on nav and screen width.
- Split icon styles between compressed and expanded.
- Add missing markup for `s-portal-nav` sample HTML.

wesleyboar committed:
Quick: Fix inaccurate CSS test case

wesleyboar committed:
GH-101: Portal Nav Mobile - Use Mixin

- New CSS plugin test for nested @extend's.
- New mixin "mobile button".
- Use mixin for mobile button styles in header.

wesleyboar committed:
Merge pull request #239 from TACC/task/GH-101-header-redesign--portal-nav-mobile

wesleyboar committed:
Quick: GH-101: More accurate comment

wesleyboar committed:
GH-101: Portal Nav Mobile - Layout (Flex to Grid)

- Confert "mobile button" layout from flex to grid.
- Update "mobile button" instances accordingly.

wesleyboar committed:
Hotfix: GH-101: Portal Nav Mobile (Grid Button)

Do not set `display: grid` in `x-mobile-button`.

wesleyboar committed:
GH-101: Portal Nav Mobile - Layout (Menu Pushdown)

- Add more nav media queries.
- Use new constants.
- Shove Portal Nav up into Search Bar row (do not properly share 1 row).
- Ensure icon spacing is accurate given different nav states.
- Add background for Portal Nav & Search Bar "row".
- Make all mobile dropdown menu items have same font size.
- Retain wide portal nav for one breakpoint within collapsed navbar.
- Limit Portal Nav load transition to when navbar is expanded.
- Add new constants.

wesleyboar committed:
Merge pull request #241 from TACC/task/GH-101-header-redesign--portal-nav-mobile

wesleyboar committed:
GH-101: Cleanup

- __New__: Migrate `s-header-….css` into to `s-header.css`.
- _Minor_: Clearer comment in `c-branding`.
- _Minor_: Migrate style specific to `tacc-search-bar` to its CSS file.
- __Fix__: Resolve Docs bugs (caused by outdated Docs styles).
- _Minor_: Clearer comments in `s-cms-nav`.
- __New__: Refactor `s-header.css`.
- _Minor_: Clearer comments in `s-portal-nav`.

wesleyboar committed:
Merge pull request #242 from TACC/task/GH-101-header-redesign--cleanup

GH-101: Cleanup

wesleyboar committed:
GH-101: Search (Initial Commit)

WARNING: Not tested on light theme, yet.

- __Fix__: Remove "freeze variables".
- __New__: Remove old, bad search styles.
- __New__: Add new, good search styles.
- _Minor_: Remove beginner comments about `position: relative`.
- __New__: Add new utility for hiding elements.
- __New__: Add new theme/env vars for search border and background.
- __Fix__: Simplify search field markup.

wesleyboar committed:
GH-101: Search (Frontera Cleanup)

Remove Frontera styles that are no longer necessary.

wesleyboar committed:
Merge pull request #244 from TACC/task/GH-101-header-redesign--search-bar

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Quick: GH-101: New Test: Custom Selector Constant

A custom selector is constant.

One can not change a custom selector at runtime.

wesleyboar committed:
Hotfix: GH-101: Match Portal+Docs Nav Breakpoint

Use `navbar-expand-lg` (for now) to match Portal & Docs.

Notes:
- Ideally CSS can change media queries to match diff markup per subsite.
- But there is NOT feasible CSS way to do this.
- And the effort for a way it can be done (nesting rules) is too great.
- So great an effort I might as well update Portal+Docs markup.

wesleyboar committed:
Merge pull request #247 from TACC/hotfix/GH-101-header-redesign--media-queries

wesleyboar committed:
GH-101: Docs Side Nav to not cover Head Mobile Nav

The Docs nav was covering the Header nav because of its z-index.

Older use of z-index required excess application and values.

Since browsers agreed on stacking context, z-index usage is simplified.

wesleyboar committed:
GH-101: Must Not Apply CMS-Only CSS to Portal+Docs

Explanation:
- The selector `.s-header:not(.is-on-portal)` failed.
- Because It affected Portal & Docs, which lack `.is-on-portal`.
- It must be limited to CMS (until Portal & Docs have `.is-on-portal`).

Other Changes:
- Improved related comments.
- Added and used `--nav-x-wide-and-...` custom media queries.

wesleyboar committed:
GH-101: Fix Portal Nav Width Issues

- _Minor_: Define variable (even if its absense breaks nothing).
- __Fix__: Remove effort to half-match design.
- _Minor_: Accept and note the deviation above.

wesleyboar committed:
GH-101: Fix Nav & Dropdown Spacing

- __Fix__: Spacing values were wrong.
- _Minor_: Reorganized Dropdown styles to mirror Nav style organization.

wesleyboar committed:
GH-101: Fix Nav Item Color When Dropdown is Shown

- __Fix__: No background color for shown nav item dropdown.
- _Minor_: Add missing comments for "Nav Link: State: Mobile/Desktop".

wesleyboar committed:
GH-101: Add Top Space for Mobile Navbar & CMS Nav

wesleyboar committed:
GH-101: Fix Left & Right Spacing for Navbar

- __Fix__: Make this spacing more match design.
- __New__: Propogate constant renames and use new constants.
- __New__: Distinguish `--header-navbar-...-pad` for mobile & normal.

wesleyboar committed:
GH-101: Improve Portal Nav Dropdown Icon Position

wesleyboar committed:
GH-101: Add Fallback for `backdrop-filter`

- __New__: Add CSS nesting support.
- __New__: Add CSS nesting test case.
- __Fix__: Darken background for browsers that cannot blur yet.
- _MInor_: Fix other `backdrop-filter` instances.

wesleyboar committed:
GH-101: Fix login link. Polish username link.

wesleyboar committed:
Merge pull request #248 from TACC/bugfix/GH-101-header-redesign--misc

Strmiska commented on taccsite_cms/static/site_cms/css/src/_themes/export.js:
```diff
@@ -0,0 +1,38 @@
+ */
+const merge = require('merge-lite');
+const rootPath = __dirname + '/../../../../../../';
+const settings = require( rootPath + 'taccsite_cms/settings.json');
```
If we need to reference this path anywhere else in the CMS it might be worth moving this path into `settings`. Then we could set the path like like `__dirname + settings_path`. An alternative approach would be to create an alias for the path if that's an option.

This is just a suggestion though.

tacc-wbomar replied:
**Thank you.** I have referenced this suggestion at [PR #192](https://github.com/TACC/Core-CMS/pull/192#discussion_r649229084) (GH-101/#222 is piggy-backing off that isolated but un-merged feature). Your comment has made me look harder at the feature. It can be improved. If you are willing to assist further, please see [the current status of the feature](https://github.com/TACC/Core-CMS/pull/192#issuecomment-858704155).

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Quick: FP-526 (complete) → FP-636 (someday)

wesleyboar committed:
GH-101: Remove pre-FP-526 Portal Nav icon styles

wesleyboar committed:
Merge branch 'task/GH-191-support-light-navbar-for-dark-logos' into task/GH-101-header-redesign

wesleyboar committed:
GH-101: Add post-FP-526 Portal Nav icon styles

These styles are temporary.

wesleyboar committed:
GH-101: Prevent wrap "Log In" as Portal Nav slides

Prevent the "Log In" text from wrapping when the Portal Nav slides in.

wesleyboar committed:
GH-101: Narrow Portal Nav as Dropdown not Pushdown

On narrow screens:
- Make the Portal Nav be a dropdown (again).
- Do not let the Portal Nav pushdown CMS menu—cover it instead.

wesleyboar committed:
GH-101: Portal Nav Layout Only Expanded or Compressed

- Remove the distinction between two narrow layouts.
- Just have one collapsed nav layout.

wesleyboar committed:
GH-101: Portal Nav Align with Screen not Parent

Notes:
- Portal Only (CMS and Docs do not have tis problem)
- New file for Portal-specific code.

What:
- Align Portal Nav toggle to distance from right of screen, not parent.
- Align Portal Nav menu to distance from right of screen, not parent.

Why:
- If aligned to right of parent, then it could appear off screen.
- Because parent width matches body, which has min-width on Portal.
- The screen may be narrower than body for certain Portal sections.

wesleyboar committed:
GH-101: Portal Nav Width Always Matches Parent

On any screen width, nav dropdown width to be 100% of parent width.

wesleyboar committed:
GH-101: Fix CMS Dropdown Font-Size Too Small

wesleyboar committed:
GH-101: Equal Size Icons. Fix Dropdown Item Height

- All header icons should always be the same size.
- The dropdown item vert padding is slightly talller (to reach ~25px).
- The dropdown item vert padding is now set correctly with variable.

wesleyboar committed:
GH-101: Fix Docs Image Overwrites

wesleyboar committed:
GH-101: Fix Docs Icon Font Styling

Fixes for CMS `task/GH-101` with `dev` Docs:
- Hide all .sr-only instances
- Overwrite TACC icon styles affecting Body
- Overwrite ReadTheDocs/FontAwesome icon styles affecting Header

wesleyboar committed:
GH-101: Portal & Docs Narrow Screen Header Fixes

wesleyboar committed:
GH-101: Fix Unfocusable Search

wesleyboar committed:
GH-101: Add Space to Right of Standalone CMS Nav

- Add space between standalone CMS nav and right side of screen.
- Change value of and then use constant `--header-navbar-normal-right-pad`.
- Better commenting for constant `--header-navbar-...-...-pad`.

wesleyboar committed:
GH-101: Do not install plugin already within other

Plugin postcss-preset-env includes all plugins for enabeld features.

wesleyboar committed:
GH-101: Use `:not` selector w/ multiple selectors

I added this feature in GH-101, but I forgot to use it.

wesleyboar committed:
GH-101: Delete file unsued since faab92b

wesleyboar committed:
GH-101: Use x-hide's for u-hide's. Rename one.

wesleyboar committed:
GH-101: Prevent wrap of portal nav toggle

wesleyboar committed:
GH-101: Rename --navbar-button to --navbar-toggle

wesleyboar committed:
GH-101: Prevent css nano from removing units

Relevant bugs:
https://github.com/cssnano/cssnano/issues/286
https://github.com/cssnano/cssnano/issues/229
https://github.com/cssnano/cssnano/issues/327

wesleyboar committed:
GH-101: Add note about when some Docs code can die

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
GH-101: Update submodule to merge of main

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Quick: Update taccsite_custom to latest

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'task/GH-191-support-light-navbar-for-dark-logos' into task/GH-101-header-redesign

wesleyboar committed:
GH-101: Noop: Update comment to be consistent

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'task/GH-191-support-light-navbar-for-dark-logos' into task/GH-101-header-redesign

wesleyboar committed:
Quick: Update submod to last commit on its branch

wesleyboar committed:
GH-101: Build settings.json before CSS

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'task/GH-191-support-light-navbar-for-dark-logos' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign

wesleyboar committed:
Merge branch 'main' into task/GH-101-header-redesign
